packages/kbn-test/src/functional_tests/cli/start_servers/args.js

```diff
@@ -0,0 +1,111 @@
+import chalk from 'chalk';
+import dedent from 'dedent';
+import { createToolingLog, pickLevelFromFlags } from '@kbn/dev-utils';
+
+export const options = {
+  help: { desc: 'Display this menu and exit.' },
+  config: {
+    arg: '<file>',
+    desc: 'Pass in a config',
+  },
+  'es-from': {
+    arg: '<snapshot|source>',
+    choices: ['snapshot', 'source'],
+    desc: 'Build Elasticsearch from source or run from snapshot.',
+    default: 'snapshot',
+  },
+  'kibana-install-dir': {
+    arg: '<dir>',
+    desc: 'Run Kibana from existing install directory instead of from source.',
+  },
+  verbose: { desc: 'Log everything.' },
+  debug: { desc: 'Run in debug mode.' },
+  quiet: { desc: 'Only log errors.' },
+  silent: { desc: 'Log nothing.' },
+};
+
+export function displayHelp() {
+  const helpOptions = Object.keys(options)
+    .filter(opt => opt !== '_')
+    .map(name => {
+      return {
+        ...options[name],
+        usage: `${name} ${options[name].arg || ''}`,
+        default: options[name].default
+          ? `Default: ${options[name].default}`
+          : '',
+      };
+    })
+    .map(option => {
+      return `--${option.usage.padEnd(28)} ${option.desc} ${option.default}`;
+    })
+    .join(`\n      `);
+
+  console.log(
+    dedent(`
+    Run Functional Tests
+
+    Usage:
+      node scripts/functional_tests_server --help
+      node scripts/functional_tests_server [--config <file>]
+      node scripts/functional_tests_server [options] [-- --<other args>]
+
+    Options:
+      ${helpOptions}
+    `)
+  );
+}
+
+export function processOptions(userOptions, defaultConfigPath) {
+  // Don't validate if help passed
+  if (userOptions.help) return userOptions;
+
+  validateOptions(userOptions);
+
+  const config = userOptions.config || defaultConfigPath;
+
+  if (!config) {
+    console.log(chalk.red(`functional_tests_server: config is required`));
+    process.exit(9);
```

Comment:
I agree we shouldn't have process.exit spread across a lot of files, but i created the /cli folders so that all the cli-related things would be in one place. I feel like it's fair to expect control flow things here, but I can do some error handling organization so that exits happen at the highest level only.

As for exit codes, since this is a node app, I felt we should follow the exit codes specified by the library. `9` is appropriate for params being wrong. But anyway, why do you think we couldn't use the right code when throwing an exception here and having the higher function `process.exit`?

Reply:
Looking at where process.exit is happening atm, 
```sh
☁  Jun 5 17:31:11  kibana [archanid-18593] rg -- 'process.exit' packages/kbn-test/
packages/kbn-test/src/functional_tests/cli/run_tests/args.js
99:      process.exit(9);
123:      process.exit(9);
139:      process.exit(9);

packages/kbn-test/src/functional_tests/cli/start_servers/args.js
88:    process.exit(9);
111:      process.exit(9);
127:      process.exit(9);

packages/kbn-test/src/functional_tests/cli/run_tests/cli.js
44:    process.exit(1);

packages/kbn-test/src/functional_tests/cli/start_servers/cli.js
41:    process.exit(1);
```

I guess one nice thing is that all the arg/param related things are together. I'm just weighing the tradeoffs.

Reply:
> As for exit codes, since this is a node app, I felt we should follow the exit codes specified by the library. `9` is appropriate for params being wrong. But anyway, why do you think we couldn't use the right code when throwing an exception here and having the higher function `process.exit`?

I think generally we _should_ use the right code when we throw an exception whenever possible. I just didn't know where you got 9 from as I was always thinking that [all these codes](https://nodejs.org/api/process.html#process_exit_codes) are reserved by NodeJS runtime itself to easily distinguish V8 runtime and user program errors if needed. And I had an impression that `Invalid Argument` was about NodeJS/V8 command line arguments. I can't find anything off-hand that would support my assumption, so I may be totally wrong here and feel free to ignore me :)

Reply:
I went ahead and removed the error code 9.